cmd/ipfs/daemon.go

```diff
@@ -452,8 +429,8 @@ func serveHTTPGateway(req cmds.Request) (error, <-chan error) {
 	if err != nil {
 		return fmt.Errorf("serveHTTPGateway: req.Option(%s) failed: %s", writableKwd, err), nil
 	}
-	if !writableOptionFound {
-		writable = cfg.Gateway.Writable
+	if writableOptionFound {
+		cfg.Gateway.Writable = writable
```

Comment:
Mh okay, we shouldn't be using `Default(false)` with this one. From what I can tell, `--writable` *if present* is supposed to override `cfg.Gateway.Writable`.